### PR TITLE
mavlink: fix yawing to North for LOITER_TIME

### PR DESCRIPTION
**Pull request by julianoes**
This fixes the issue where the vehicle would yaw to North for a LOITER_TIME mission item.

We ran into this issue for the Yuneec + 3DR integration. @dagar would be great if you could sanity check that.

### CODE REVIEW

File: src/modules/mavlink/mavlink_mission.cpp [hunk 1]
```diff
@@ -1335,6 +1336,21 @@ MavlinkMissionManager::parse_mavlink_mission_item(const mavlink_mission_item_t *
...
 			mission_item->loiter_exit_xtrack = (mavlink_mission_item->param4 > 0);
+
+			// The spec says:
+			// Forward moving aircraft this sets exit xtrack location:
+			// 0 for center of loiter wp, 1 for exit location.
+			// Else, this is desired yaw angle.
+			//
+			// Therefore, we don't touch yaw by setting it NAN for these two cases.
+			if (fabsf(mavlink_mission_item->param4 - 0.0f) < FLT_EPSILON ||
+			    fabsf(mavlink_mission_item->param4 - 1.0f) < FLT_EPSILON) {
```
Comment (dagar) [inline src/modules/mavlink/mavlink_mission.cpp]: I think this needs to be done by vehicle type. What if a multicopter wants a desired yaw angle of 0 or 1?

Comment (julianoes) [reply src/modules/mavlink/mavlink_mission.cpp]: Ah, just enter 360 :smile:

**Comment by julianoes** [comment]
@dagar ah, now I understand what the spec means! I thought it means 0, 1, else. But it means "forward moving aircrag, else" :smile:!

**Comment by julianoes** [comment]
@dagar does fixedwing respect `mission_item.yaw` or can we set it anyway? I don't see if we know whether it's a multicopter of fixedwing in the mavlink module.

**Comment by dagar** [comment]
No it doesn't.

**Comment by dagar** [comment]
Note - need to check the other loiters as well.

**Comment by julianoes** [comment]
> No it doesn't.

Ok, so I'll just set yaw for all loiters then.

**Comment by julianoes** [comment]
@dagar please review again.

**Pull request merged**